viewing_party/party.py

```diff
@@ -1,23 +1,127 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    movie_dict = {}
+    if title and genre and rating:
+        movie_dict["title"] = title   
+        movie_dict["genre"] = genre        
+        movie_dict["rating"] = rating
+    else: 
+        return None    
+    return movie_dict
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for item in user_data["watchlist"]:
+        if item["title"] == title:
+            user_data["watched"].append(item)
+            user_data["watchlist"].remove(item)
+    return user_data

```

Comment:
This function works, but as an aside it contains an "anti-pattern" that is important to be aware of.  An element of the list that is being iterated over is being removed inside the for loop.  This won't cause problems in this particular case, but consider this slight change:
```
def watch_movie(user_data, title):
    for i in range(len(user_data["watchlist"])):
        item = user_data["watchlist"][i]
        if item["title"] == title:
            user_data["watched"].append(item)
            user_data["watchlist"].remove(item)
    return user_data
```
If we were to test this in a scenario where `user_data["watchlist"]` has 2 elements, and the first is the movie that we are moving to `user_data["watched"]`, the second iteration through the loop will throw an index error, because it will try to access the 2nd element in `user_data["watchlist"]`, but there no longer is a second element.  

As a coding pattern this is one that I recommend avoiding, because it has so much potential for unforeseen errors.  One fix is to return immediately after making the swap (ie, move line 26 inside the if block so that the loop ends as soon as the swap is made).  Another fix is use the loop to find the item, but not make the swap until after the loop ends.